### PR TITLE
NR-229763 Add UUID-like header to extensions on all api_server requests

### PR DESCRIPTION
**Pull request by dsellarsnr**

### CODE REVIEW

File: shared/agent/src/api/codestream/codestreamApi.ts [hunk 1]
```diff
@@ -384,7 +386,10 @@ export class CodeStreamApiProvider implements ApiProvider {
 		private readonly _version: VersionInfo,
 		private readonly _httpsAgent: HttpsAgent | HttpsProxyAgent<string> | HttpAgent | undefined,
 		private readonly _strictSSL: boolean
-	) {}
+	) {
+		this._clientId = machineIdSync();
```
Comment (bcanzanella) [inline shared/agent/src/api/codestream/codestreamApi.ts:390]: any chance it could throw?

Comment (dsellarsnr) [reply shared/agent/src/api/codestream/codestreamApi.ts:390]: Yeah, what should we set it to if it throws - blank?

**Review by CalvinAllen — APPROVED** [review]
Pending @bcanzanella's comment

File: shared/agent/src/api/codestream/codestreamApi.ts (hunk 1 above)
Comment (bcanzanella) [reply shared/agent/src/api/codestream/codestreamApi.ts:390]: sounds good

**Pull request merged**